liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/DevMojo.java

```diff
@@ -637,9 +637,14 @@ private void injectTestId(Xpp3Dom config) {
         return elements.toArray(new Element[elements.size()]);
     }
 
-    private void runMojo(String plugin, String goal, String serverName, List<String> dependencies)
+    private void runMojo(String groupId, String artifactId, String goal, String serverName, List<String> dependencies)
             throws MojoExecutionException {
-        Plugin mavenPlugin = project.getPlugin(plugin);
+        Plugin mavenPlugin = project.getPlugin(Plugin.constructKey(groupId, artifactId));
+        if (mavenPlugin == null) {
+            mavenPlugin = new Plugin();
+            mavenPlugin.setGroupId(groupId);
+            mavenPlugin.setArtifactId(artifactId);
```

Comment:
It seems not working without including the plugin version.

```
aused by: java.lang.NullPointerException
    at org.apache.maven.plugin.DefaultPluginDescriptorCache$CacheKey.<init> (DefaultPluginDescriptorCache.java:168)
    at org.apache.maven.plugin.DefaultPluginDescriptorCache.createKey (DefaultPluginDescriptorCache.java:63)
    at org.apache.maven.plugin.internal.DefaultMavenPluginManager.getPluginDescriptor (DefaultMavenPluginManager.java:175)
    at org.apache.maven.plugin.DefaultBuildPluginManager.loadPlugin (DefaultBuildPluginManager.java:86)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.twdata.maven.mojoexecutor.MavenCompatibilityHelper.loadPluginDescriptor (MavenCompatibilityHelper.java:86)
    at org.twdata.maven.mojoexecutor.MojoExecutor.executeMojo (MojoExecutor.java:110)
    at net.wasdev.wlp.maven.plugins.server.DevMojo.runMojo (DevMojo.java:649)
    at net.wasdev.wlp.maven.plugins.server.DevMojo.doExecute (DevMojo.java:442)
    at org.codehaus.mojo.pluginsupport.MojoSupport.execute (MojoSupport.java:122)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
```

Reply:
Thanks for noticing that.  The error occurs if `liberty-maven-app-parent` is not included as a parent.

Reply:
Fixed with https://github.com/ericglau/ci.maven/commit/93166a29e895e176d1d8553a7a20afb458aac8eb